kotlin_client/src/main/java/com/github/paylike/kotlin_client/dto/HintsResponse.kt

```diff
@@ -0,0 +1,9 @@
+package com.github.paylike.kotlin_client.dto
+
+// Describes the hints array received after executing a challenge successfully.
+class HintsResponse(json: Map<String, Any>){
```

Comment:
Also `json` is going to be `JsonObject` in your case as `kotlinx.serialization` works with that class

Reply:
This is true for all of the DTOs pretty much, every map (e.g. `{ "foo": "bar" }` will be `JsonObject`, every list will become `JsonArray` and every value is `JsonPrimitiveValue` if I remember correctly 😄 But I think this is not the de facto fastest way to do this, so I assume you could also use annotations or something to generate auto deserializers for classes. Despite that, I prefer it this way so we can explicitly see what is going on.